module/Decision/src/Decision/Service/Decision.php

```diff
@@ -523,4 +598,23 @@ public function isAllowedToBrowseFiles()
     {
         return $this->isAllowed('browse', 'files');
     }
+
+    /**
+     * Checks the user's permission
+     *
+     * @param string $operation
+     * @param string $resource
+     * @param string $errorMessage English error message
+     * @throws NotAllowedException If the user doesn't have permission
+     */
+    private function isAllowedOrFail($operation, $resource, $errorMessage)
```

Comment:
I kind of like the idea of having a method for this, as this is a very regularly used block of code.

Reply:
Maybe refactoring it in a [trait](https://www.php.net/manual/en/language.oop5.traits.php)? Other services or controllers can then make use of this shared functionality.

Reply:
Well, if you would like other services to make use of it, you would have to refactor that code as well, which is a lot of work. I'm not quite sure what a good place would be to put it. But for this PR, it's fine like this.